lib/bdev/nvme/blockdev_nvme.c

```diff
@@ -311,6 +391,8 @@ blockdev_nvme_create_cb(void *io_device, uint32_t priority, void *ctx_buf, void
 
 	spdk_poller_register(&ch->poller, blockdev_nvme_poll, ch->qpair,
 			     spdk_app_get_current_core(), NULL, 0);
+	spdk_poller_register(&ch->adminq_poller, blockdev_nvme_poll_adminq, ctrlr,
```

Comment:
This will cause an admin queue poller to be registered for each I/O channel, but there is only one admin queue per NVMe controller. This will lead to poor performance due to the locking required, and I don't think the current implementation in this patch will behave correctly when used with multiple I/O channels per controller.

I believe it would be better to have a single admin queue poller per NVMe controller that should process all admin completions and use the SPDK event library function spdk_event_call() to pass the completion back to the correct thread.  This admin poller could be registered in attach_cb().